Doc/library/compression.zstd.rst

```diff
@@ -0,0 +1,780 @@
+   in which case the named file is opened, or it can be an existing file object
+   to read from or write to.
+
+   The mode argument can be either ``'r'`` for reading (default), ``'w'`` for
+   overwriting, 'a' for appending, or ``'x'`` for exclusive creation. These can
+   equivalently be given as ``'rb'``, ``'wb'``, ``'ab'``, and ``'xb'`` respectively. You may
+   also open in text mode with ``'rt'``, ``'wt'``, ``'at'``, and ``'xt'`` respectively.
+
+   When opening a file for reading, the *options* argument can be a dictionary
+   providing advanced decompression parameters; see
+   :class:`DecompressionParameter` for detailed information about supported
+   parameters. The *zstd_dict* argument is a :class:`ZstdDict` instance to be
+   used during decompression. When opening a file for reading, the *level*
+   argument should not be used.
+
+   When opening a file for writing, the *options* argument can be a dictionary
+   providing advanced decompression parameters; see
+   :class:`CompressionParameter` for detailed information about supported
+   parameters. The *level* argument is the compression level to use when
+   writing compressed data. Only one of *level* or *options* may be passed. The
+   *zstd_dict* argument is a :class:`ZstdDict` instance to be used during
+   compression.
+
+   In binary mode, this function is equivalent to the :class:`ZstdFile`
+   constructor: ``ZstdFile(file, mode, ...)``. In this case, the
+   *encoding*, *errors*, and *newline* parameters must not be provided.
+
+   In text mode, a :class:`ZstdFile` object is created, and wrapped in an
+   :class:`io.TextIOWrapper` instance with the specified encoding, error handling
+   behavior, and line endings.
+
+
+.. class:: ZstdFile(file, /, mode='r', *, level=None, options=None, zstd_dict=None)
+
+   Open a Zstandard-compressed file in binary mode.
+
+   A :class:`ZstdFile` can wrap an already-open :term:`file object`, or operate
+   directly on a named file. The *file* argument specifies either the file
+   object to wrap, or the name of the file to open (as a :class:`str`,
+   :class:`bytes` or :term:`path-like <path-like object>` object). If
+   wrapping an existing file object, the wrapped file will not be closed when
+   the :class:`ZstdFile` is closed.
+
+   The *mode* argument can be either ``"r"`` for reading (default), ``"w"`` for
+   overwriting, ``"x"`` for exclusive creation, or ``"a"`` for appending. These
+   can equivalently be given as ``"rb"``, ``"wb"``, ``"xb"`` and ``"ab"``
+   respectively.
+
+   If *file* is a file object (rather than an actual file name), a mode of
+   ``"w"`` does not truncate the file, and is instead equivalent to ``"a"``.
+
+   When opening a file for reading, the *options* argument can be a dictionary
+   providing advanced decompression parameters, see
+   :class:`DecompressionParameter` for detailed information about supported
+   parameters. The *zstd_dict* argument is a :class:`ZstdDict` instance to be
```

Comment:
Couldn't lines 59-62, 66-69, 102-105 and 109-112 be combined into a single clause, leaving only the specifics for reading and writing as unique? At the very least, four of these clauses in a short space seems a lot, two may be okay.

Reply:
Sorry I think the line numbers have changed and Github does not do a good job at showing enough context for your comment, what were you proposing?

Reply:
Yeah, line #s are always unsafe to add to comments, sorry.  It just sounded quite repetitive to, four times, say "When opening a file for... " , followed by the stanza about passing a dict of parameters. It doesn't harm anything, and if there's no obvious way to factor out some of that, then nevermind (I know they are all different combinations:  read/text/decompress, read/bytes/decompress, write/text/compress, write/bytes/compress)